source/Calamari.Common/Features/Discovery/AccountAuthenticationDetails.cs

```diff
@@ -9,15 +9,18 @@ public class AccountAuthenticationDetails<TAccountDetails> : ITargetDiscoveryAut
     {
         [JsonConstructor]
         public AccountAuthenticationDetails
-            (string type, string accountId, TAccountDetails accountDetails)
+            (string type, string accountId, string authenticationMethod, TAccountDetails accountDetails)
```

Comment:
was the `AuthenticationMethod` and existing property on the JSON?

Reply:
Yep, it's set in server just not in Calamari.

Reply:
The tests didn't include this, which I updated and it does cause failures if unspecified now. I spoke to Scott yesterday and he was confident that it's there and always has been but just wasn't set in Calamari. I can play it safe and just have Service principal as a fall back if the authentication method is not set though